app/static/js/highlight-nav-links.js

```diff
@@ -0,0 +1,44 @@
+const sections = document.querySelectorAll("section[id]")
+const navElements = document.querySelectorAll(".nav-element")
+let inViewId = ""
```

Comment:
Is this "global" variable here needed? I don’t fully understand why it’s needed, since `updateNavigationView()` is called with a specific parameter anyway. 🤔